superset/assets/src/components/OmniContainer.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/translation';
+import { SupersetClient } from '@superset-ui/connection';
+
+import Omnibar from 'omnibar';
+
+const getDashboards = query =>
+  // todo: Build a dedicated endpoint for dashboard searching
+  // i.e. superset/v1/api/dashboards?q=${query}
+   SupersetClient.get({
```

Comment:
I feel we should move network logic into a separate say `/api/network.ts` file and make proper async actions for triggering them instead of stuffing things in the containers. I've found `redux-saga` to be a wonderful library for managing async actions (i.e. having `saga`s for async actions and `reducer`s for plain action). 

@kristw @williaster what do you guys think?

Btw, if we like the idea. I'm happy to refactor one of the existing containers to get things set up.